sensing/autoware_pointcloud_preprocessor/src/outlier_filter/polar_voxel_outlier_filter_node.cpp

```diff
@@ -0,0 +1,1087 @@
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <sensor_msgs/point_cloud2_iterator.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+namespace autoware::pointcloud_preprocessor
+{
+
+template <typename... T>
+bool all_finite(T... values)
+{
+  return (... && std::isfinite(values));
+}
+
+static constexpr double diagnostics_update_period_sec = 0.1;
+static constexpr size_t point_cloud_height_organized = 1;
+
+PolarVoxelOutlierFilterComponent::PolarVoxelOutlierFilterComponent(
+  const rclcpp::NodeOptions & options)
+: Filter("PolarVoxelOutlierFilter", options), updater_(this)
+{
+  radial_resolution_m_ = declare_parameter<double>("radial_resolution_m");
+  azimuth_resolution_rad_ = declare_parameter<double>("azimuth_resolution_rad");
+  elevation_resolution_rad_ = declare_parameter<double>("elevation_resolution_rad");
+  voxel_points_threshold_ = static_cast<int>(declare_parameter<int64_t>("voxel_points_threshold"));
+  min_radius_m_ = declare_parameter<double>("min_radius_m");
+  max_radius_m_ = declare_parameter<double>("max_radius_m");
+  visibility_estimation_max_range_m_ =
+    declare_parameter<double>("visibility_estimation_max_range_m", 20.0);
+  use_return_type_classification_ = declare_parameter<bool>("use_return_type_classification", true);
+  enable_secondary_return_filtering_ = declare_parameter<bool>("filter_secondary_returns");
+  secondary_noise_threshold_ =
+    static_cast<int>(declare_parameter<int64_t>("secondary_noise_threshold"));
+  visibility_estimation_max_secondary_voxel_count_ = static_cast<int>(
+    declare_parameter<int64_t>("visibility_estimation_max_secondary_voxel_count", 0));
+  visibility_estimation_only_ = declare_parameter<bool>("visibility_estimation_only", false);
+  publish_noise_cloud_ = declare_parameter<bool>("publish_noise_cloud", false);
+
+  auto primary_return_types_param = declare_parameter<std::vector<int64_t>>("primary_return_types");
+  primary_return_types_.clear();
+  primary_return_types_.reserve(primary_return_types_param.size());
+  for (const auto & val : primary_return_types_param) {
+    primary_return_types_.push_back(static_cast<int>(val));
+    RCLCPP_DEBUG(get_logger(), "primary_return_types_ value: %d", static_cast<int>(val));
+  }
+
+  visibility_error_threshold_ = declare_parameter<double>("visibility_error_threshold", 0.5);
+  visibility_warn_threshold_ = declare_parameter<double>("visibility_warn_threshold", 0.7);
+  filter_ratio_error_threshold_ = declare_parameter<double>("filter_ratio_error_threshold", 0.5);
+  filter_ratio_warn_threshold_ = declare_parameter<double>("filter_ratio_warn_threshold", 0.7);
+  intensity_threshold_ = declare_parameter<uint8_t>("intensity_threshold", 1);
+
```

Comment:
Default params should be defined only in the config file and schema, not in C++ code according to Autoware guidelines.
Please remove them here.

If you don't want to force users to set all params in the config file, even when they are not used, you can declare them conditionally here (e.g. `if (feature_a_enabled) { declare_param("feature_a_param"); }`).

Reply:
Addressed in #00b9120274

Reply:
Just for confirmation, are the below boolean defaults left here on purpose? If yes, is there a strong reason for it?

https://github.com/autowarefoundation/autoware_universe/blob/cd9286c5203bc9143123982c097ce4b04ecf2e85/sensing/autoware_pointcloud_preprocessor/src/outlier_filter/polar_voxel_outlier_filter_node.cpp#L74

https://github.com/autowarefoundation/autoware_universe/blob/cd9286c5203bc9143123982c097ce4b04ecf2e85/sensing/autoware_pointcloud_preprocessor/src/outlier_filter/polar_voxel_outlier_filter_node.cpp#L80-L81

Reply:
Oops no, I'll fix that too!

Reply:
Should be fixed in https://github.com/autowarefoundation/autoware_universe/pull/10996/commits/2378c01835c92fea40dd8e3cf580ca0bb366a1c7